src/analysis/processing/qgsalgorithmgrid.cpp

```diff
@@ -0,0 +1,527 @@
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+//Disclaimer:This feature was developed by: Michael Minn, 2010
+
+#include "qgsalgorithmgrid.h"
+#include "qgswkbtypes.h"
+#include "qgsvectorlayer.h"
+
+///@cond PRIVATE
+
+QString QgsGridAlgorithm::name() const
+{
+  return QStringLiteral( "creategrid" );
+}
+
+QString QgsGridAlgorithm::displayName() const
+{
+  return QObject::tr( "Create Grid" );
+}
+
+QStringList QgsGridAlgorithm::tags() const
+{
+  return QObject::tr( "grid,lines,polygons,vector,create,fishnet,diamond,hexagon" ).split( ',' );
+}
+
+QString QgsGridAlgorithm::group() const
+{
+  return QObject::tr( "Vector creation" );
+}
+
+QString QgsGridAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorcreation" );
+}
+
+void QgsGridAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  //add selection parameter for grid type
+  addParameter( new QgsProcessingParameterEnum( QStringLiteral( "TYPE" ), QObject::tr( "Grid type" ), QStringList() << QObject::tr( "Point" ) << QObject::tr( "Line" ) << QObject::tr( "Rectangle (Polygon)" ) << QObject::tr( "Diamond (Polygon)" ) << QObject::tr( "Hexagon (Polygon)" ), false, 0 ) );
+
+  //add extent parameter for grid
+  addParameter( new QgsProcessingParameterExtent(QStringLiteral("EXTENT"), QObject::tr( "Grid extent" )));
+
+  //add Distance Parameter for horizontal and vertical spacing
+  addParameter( new QgsProcessingParameterDistance( QStringLiteral( "HSPACING"), QObject::tr( "Horizontal Spacing" ), 1, QString( "CRS" ), false, 0, 1000000000.0));
+  addParameter( new QgsProcessingParameterDistance( QStringLiteral( "VSPACING"), QObject::tr( "Vertical Spacing" ), 1, QString( "CRS" ), false, 0, 1000000000.0));
+
+  //add Distance Parameter for horizontal and vertical overlay
+  addParameter( new QgsProcessingParameterDistance( QStringLiteral( "HOVERLAY"), QObject::tr( "Horizontal Overlay" ), 0, QString( "CRS" ), false, 0, 1000000000.0));
+  addParameter( new QgsProcessingParameterDistance( QStringLiteral( "VOVERLAY"), QObject::tr( "Vertical Overlay" ), 0, QString( "CRS" ), false, 0, 1000000000.0));
+
+  //add Coordinate Reference System parameter
+  addParameter( new QgsProcessingParameterCrs( QStringLiteral( "CRS" ), QObject::tr( "Grid CRS" ), QStringLiteral( "ProjectCrs" )));
+
+  //add FeatureSink as output parameter
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr("Grid"), QgsProcessing::TypeVectorPolygon ) );
+}
+
+QString QgsGridAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm creates a grid layer of different shapes (rectangle, diamond, hexagon).\n\n"
```

Comment:
You can grab the existing help string from here: https://github.com/qgis/QGIS/blob/master/python/plugins/processing/algs/help/qgis.yaml#L103 (and ideally remove the paragraph from the qgis.yaml file too!)